src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverDetailsVisitor.java

```diff
@@ -0,0 +1,49 @@
+package edu.kis.powp.jobs2d.drivers.visitor;
+
+import edu.kis.powp.jobs2d.commons.Pair;
+import edu.kis.powp.jobs2d.drivers.*;
+import edu.kis.powp.jobs2d.features.RecordFeature;
+
+public class DriverDetailsVisitor implements DriverVisitor {
+    private StringBuilder stringBuilder = new StringBuilder();
+    public void clear() { stringBuilder = new StringBuilder(); }
+    public String getDetails(IDriver driver) {
+        clear();
+        driver.accept(this);
+        return stringBuilder.toString();
+    }
+
+    @Override
+    public void visit(LoggerDriver loggerDriver) {
+        Pair<Integer, Integer> position = loggerDriver.getPosition();
+        String details = "\nLoggerDriver details:" +
+                "\n Position: (" + position.getFirst() + "," + position.getSecond() + ")";
+        stringBuilder.append(details);
+    }
+
+    @Override
+    public void visit(DriversComposite driverComposite) {
+        int registeredDrivers = driverComposite.driversCount();
+        String details = "\nDriversComposite details:" +
+                "\n Registered drivers count: " + registeredDrivers;
+        stringBuilder.append(details);
+    }
+
+    @Override
+    public void visit(RecordingDriverDecorator recordingDriverDecorator) {
+        String isRecordingLabel = RecordFeature.isRecording() ? "true" : "false";
+        String details = "\nRecordingDriver details:" +
+                "\n Recording: " + isRecordingLabel +
+                "\n Serviced drivers:";
+        stringBuilder.append(details);
+    }
+
+    @Override
+    public void visit(UsageMonitorDriverDecorator usageMonitorDriverDecorator) {
```

Comment:
Similar issue decorators wrap an instance of some driver. The details should also contain information generated by DriverDetailsVisitor also for a decorated driver.